File: CloudFormation/src/test/java/com/intellij/aws/cloudformation/tests/EntityResolveTests.java
package com.intellij.aws.cloudformation.tests;

import com.intellij.aws.cloudformation.references.CloudFormationEntityReference;

public class EntityResolveTests extends ResolveTestsBase {
  public EntityResolveTests() {
    super(CloudFormationEntityReference.class);
  }

  public void testDependsOnSingle() throws Exception {
    assertEntityResolve("dependsOnSingle", "WebServerUser");
  }

  public void testDependsOnMulti() throws Exception {
    assertEntityResolve("dependsOnMulti", "WebServerUser1", "WebServerUser2");
  }

  public void testFindInMapRefToResource() throws Exception {
    assertEntityResolve("findInMapRefToResource", NotResolved);
  }

  public void testFindInMapRefToMapping() throws Exception {
    assertEntityResolve("findInMapRefToMapping", "AWSInstanceType2Arch");
  }

  public void testGetAttrParameterRef() throws Exception {
    assertEntityResolve("getAttrParameterRef", NotResolved);
  }

  public void testGetAttrResourceRef() throws Exception {
    assertEntityResolve("getAttrResourceRef", "WebServerUser");
  }

  public void testRefToResource() throws Exception {
    assertEntityResolve("refToResource", "WebServerUser");
  }

  public void testRefToParameter() throws Exception {
    assertEntityResolve("refToParameter", "WebServerInstanceType");
  }

  public void testRefToMapping() throws Exception {
    assertEntityResolve("refToMapping", NotResolved);
  }

  public void testRefToOutput() throws Exception {
    assertEntityResolve("refToOutput", NotResolved, NotResolved);
  }

  public void testConditionOnResource() throws Exception {
    assertEntityResolve("conditionOnResource", "CreateProdResources");
  }

  public void testCondition() throws Exception {
    assertEntityResolve("condition", "CreateProdResources");
  }

  public void testConditionInIf() throws Exception {
    assertEntityResolve("conditionInIf", "CreateProdResources");
  }

  @Override
  protected String getTestDataPath() {
    return TestUtil.getTestDataPath("resolve");
  }
}

File: CloudFormation/src/test/kotlin/com/intellij/aws/cloudformation/tests/EntityResolveTests.kt
package com.intellij.aws.cloudformation.tests

import com.intellij.aws.cloudformation.references.CloudFormationEntityReference

class EntityResolveTests : ResolveTestsBase(CloudFormationEntityReference::class.java) {
  fun testDependsOnSingle() {
    assertEntityResolve("dependsOnSingle.template", "WebServerUser")
  }

  fun testDependsOnMulti() {
    assertEntityResolve("dependsOnMulti.template", "WebServerUser1", "WebServerUser2")
  }

  fun testFindInMapRefToResource() {
    assertEntityResolve("findInMapRefToResource.template", NotResolved)
  }

  fun testFindInMapRefToMapping() {
    assertEntityResolve("findInMapRefToMapping.template", "AWSInstanceType2Arch")
  }

  fun testGetAttrParameterRef() {
    assertEntityResolve("getAttrParameterRef.template", NotResolved)
  }

  fun testGetAttrResourceRef() {
    assertEntityResolve("getAttrResourceRef.template", "WebServerUser")
  }

  fun testRefToResource() {
    assertEntityResolve("refToResource.template", "WebServerUser")
  }

  fun testRefToParameter() {
    assertEntityResolve("refToParameter.template", "WebServerInstanceType")
  }

  fun testRefToParameterInCloudFormationInterfaceParameterLabels() {
    assertEntityResolve("refToParameterInCloudFormationInterfaceParameterLabels.template", "WebServer12InstanceType1", "WebServer12InstanceType2", NotResolved)
  }

  fun testRefToParameterInCloudFormationInterfaceParameterGroups() {
    assertEntityResolve("refToParameterInCloudFormationInterfaceParameterGroups.template", NotResolved, "WebServerInstanceType1", "WebServerInstanceType2")
  }

  fun testRefToMapping() {
    assertEntityResolve("refToMapping.template", NotResolved)
  }

  fun testRefToOutput() {
    assertEntityResolve("refToOutput.template", NotResolved, NotResolved)
  }

  fun testConditionOnResource() = assertEntityResolve("conditionOnResource.template", "CreateProdResources")

  fun testShortRef() = assertEntityResolve("shortRef.yaml", "WebServer")

  fun testCondition() {
    assertEntityResolve("condition.template", "CreateProdResources")
  }

  fun testConditionInIf() {
    assertEntityResolve("conditionInIf.template", "CreateProdResources")
  }

  override fun getTestDataPath(): String {
    return TestUtil.getTestDataPath("resolve")
  }
}